### ISSUE TITLE
[BUG] 

### ISSUE THREAD

**Issue by Luxcis** [violentmonkey/violentmonkey#1937]
### Sequence of actions: 
同步出错，如果之前未同步过或者以同步过然后脚本删完，那是可以同步的，但是只要增减了脚本再次同步就会出错，当前尝试了onedrive、wedav都是一样的问题。

### Problem: 
尝试同步百分百报错

#### Environment:
- OS: MacOS Ventura 13.5.1 
- Browser: Edge 118.0.2088.46 (正式版本) (x86_64)
- Violentmonkey Version: 暴力猴v2.15.0

**Comment by Luxcis** [comment]
> Sounds like #1260

I don't think so. I tried it on windows11 and it worked well. May be only appears on the Mac?

**Comment by cyfung1031** [comment]
@Luxcis  You are using the stable version 2.15.0, which contains many bugs. You can install beta version 2.15.6 or the latest CI build. These issues might vanish.

**Comment by megaslx** [comment]
I'm facing the similar issue.
Environment:
    OS: Windows11
    Browser: FireFox 120
    Violentmonkey Version: 2.16 and 2.16.1beta

sync fail at the end and deleted local scripts are installed back after sync
